tests/ALCProxy.Tests/ALCProxyTests.cs

```diff
@@ -0,0 +1,268 @@
+using ALCProxy.Proxy;
+using Xunit;
+
+namespace ALCProxy.Tests
+{
+    public interface ITest
+    {
+        public string PrintContext();
+        public int DoThing2(int a, List<string> list);
+        public int DoThing3(int a, Test2 t);
+        public Test2 ReturnUserType();
+        public int SimpleMethod();
+    }
+
+    public interface IGeneric<T>
+    {
+        public string PrintContext();
+        public int DoThing2(int a, List<string> list);
+        public int DoThing3(int a, Test2 t);
+        public string DoThing4(T t);
+        public string GenericMethodTest<I>();
+    }
+
+    public class Test2
+    {
+        public int testValue;
+        public Test2()
+        {
+            testValue = 5;
+        }
+        public Test2(int start)
+        {
+            testValue = start;
+        }
+        public void DoThingy()
+        {
+            testValue++;
+        }
+    }
+    public class GenericClass<T> : IGeneric<T>
+    {
+        private readonly string _instance = "testString";
+        private T _instance2;
+        public GenericClass()
+        {
+        }
+        public GenericClass(T t)
+        {
+            _instance2 = t;
+        }
+        public string PrintContext()
+        {
+            var a = Assembly.GetExecutingAssembly();
+            return AssemblyLoadContext.GetLoadContext(a).Name;
+        }
+        public string GenericMethodTest<I>()
+        {
+            return typeof(I).ToString();
+        }
+        public int DoThing2(int a, List<string> list)
+        {
+            return _instance.Length;
+        }
+        public int DoThing3(int a, Test2 t)
+        {
+            t.DoThingy();
+            return 6;
+        }
+        public string DoThing4(T tester)
+        {
+            return tester.ToString();
+        }
+    }
+
+    public class Test : ITest
+    {
+        public string PrintContext()
+        {
+            var a = Assembly.GetExecutingAssembly();
+            return AssemblyLoadContext.GetLoadContext(a).Name;
+        }
+        public int DoThing2(int a, List<string> list)
+        {
+            Console.WriteLine(a);
+
+            return a + list[0].Length;
+        }
+        public int DoThing3(int a, Test2 t)
+        {
+            t.DoThingy();
+            return 5;
+        }
+
+        public Test2 ReturnUserType()
+        {
+            return new Test2();
+        }
+        public int SimpleMethod()
+        {
+            return 3;
+        }
+    }
+
+    public class TestObjectParamConst : ITest
+    {
+        int aa;
+        string bb;
+        Test2 tt;
+        public TestObjectParamConst(int a, string b)
+        {
+            this.aa = a;
+            this.bb = b;
+            tt = new Test2();
+        }
+        public TestObjectParamConst(int a, string b, Test2 t)
+        {
+            this.aa = a;
+            this.bb = b;
+            tt = t;
+        }
+        public int DoThing2(int a, List<string> list)
```

Comment:
I'd recommend naming these methods after what they're testing. This will make the tests easier to read.